scrabble.js

```diff
@@ -1,14 +1,149 @@
 
 const Scrabble = {
   score(word) {
+    let totalScore = 0;
+    if (typeof word !== 'string') {
+      throw "Invalid input, please enter a word!";
+    }
 
+    let scrabbleWord = word.split("");
+    if (scrabbleWord.length > 7 || scrabbleWord.length === 0) {
+      throw "Word need to have between 1-7 characters";
+    }else if (scrabbleWord.length === 7) {
+      totalScore += 50;
+    }
+
+    scrabbleWord.forEach((letter) => {
+      switch(letter.toLowerCase()) {
+        case "a":
+        case "e":
+        case "i":
```

Comment:
I'm not a big fan of organizing the letter scores in a `switch` statement like this. It's not very DRY (you've had to type `case` 26 times), but more importantly the data is intertwined with the code that uses it, which means there's no easy way to reuse this information.

For example, what if the player had a hand full of tiles, and you wanted to be able to display the value of each of those letters? You would have to copy/paste this switch in that code, or at least extract it to a method.

A better approach would be to store the scores in a hash:

```javascript
const LETTER_VALUES = {
  a: 1,
  b: 3,
  // ...
}
```

then here you could use it like this:

```javascript
word.forEach((letter) => {
  const value = LETTER_VALUES[letter.toLowerCase()];
  if (value) {
    totalScore += letter.toLowerCase();
  } else {
    throw new Error(`Invalid character ${letter}`);
  }
}
```